### DISCUSSION TITLE
Radio componments with custom colors

### DISCUSSION THREAD

**Discussion by xpierroz** [zauberzeug/nicegui#1015] [Q&A] [answered]
### Question

Hey,
I have this code
`radio2 = ui.radio({1: 'Streaming', 2: 'Playing', 3: 'Listening'}).classes('w-full center justify-center items-center flex')`

However, I want my 1 streaming value to be red (color=red-9)
My 2nd value playing to be yellow,
etc etc ...
How can I do that, because if I apply .props to the element, it's going to be for all the value (streaming, playing, listening) ...

**Comment by sergeyyurkov1** [comment]
Hello.

In Quasar, it is possible to override default styling with custom CSS. Like so:

```css
.q-radio:nth-of-type(1) > .q-radio__inner--truthy {
  color: red !important;
}
```

I tested that, and it works. However, it doesn't seem to work with NiceGUI. I tried to add custom CSS with `ui.add_head_html` and `ui.query` but to no avail.

Maybe someone who is more familiar with NiceGUI can explain what I'm doing wrong here. Anyway, this can be your starting point.

**Comment by xpierroz** [comment]
Tbh I don't think it is possible in NiceGUI, i'll try to use other componments, but it's not the best optimised, still waiting for an answer for someone who's more familiar with NiceGUI

**Comment by falkoschindler** [comment, marked as answer]
NiceGUI's `ui.radio` seems to be rendered a bit differently than what @sergeyyurkov1 tried in Quasar. Inspecting the DOM shows another div around the `.q-radio` element. So this modified CSS works:

```css
div:nth-of-type(1) > .q-radio > .q-radio__inner--truthy {
  color: red !important;
}
```

To restrict the custom colors to specific elements only, I would use a custom class like "colored-radio" for the radio element. So the final code might look something like this:

```py
ui.add_head_html('''
<style>
.colored-radio div:nth-of-type(1) > .q-radio > .q-radio__inner--truthy {
  color: red !important;
}
.colored-radio div:nth-of-type(2) > .q-radio > .q-radio__inner--truthy {
    color: green !important;
}
.colored-radio div:nth-of-type(3) > .q-radio > .q-radio__inner--truthy {
    color: blue !important;
}
</style>
''')

ui.radio(['red', 'green', 'blue']).classes('colored-radio')
```

**Reply by xpierroz** [reply]
Thanks a lot man! Worked perfectly

**Reply by petergaultney** [reply]
this is not working for me today.

I must be doing _something_ wrong, but I have confirmed three things:
1. i am successfully creating a `<style> ...CSS </style>` string and calling `ui.add_head_html(mycss)` on it.
2. I have completely restarted the application
3. I am not seeing that string anywhere in the Inspect->Elements for the page

If i add the style element manually via Chrome's inspector, then it applies as expected. It's just not showing up in what the browser receives from the server.

**Reply by falkoschindler** [reply]
@petergaultney Does the code snippet from https://github.com/zauberzeug/nicegui/discussions/1015#discussioncomment-6146190 work?

**Reply by petergaultney** [reply]
interestingly, yes. Or rather, no, but it did when I moved the CSS snippet to the same page that I was rendering the radio button (as a test) on.

Which leads me to - Is there a way to put this CSS into the `head` of _every_ page? I naively assumed this was an app-wide addition, but I see now that it is not. It would be nice to have this on every page without having to repeat myself, but that may not be possible.

**Reply by falkoschindler** [reply]
Glad to hear that you found a solution. 🙂
Setting head and body HTML on _every_ page has been implemented in PR #2127 and will be part of the upcoming release of version 1.4.6. It introduces a parameter `ui.add_head_html(..., shared: bool)` that can be used to send HTML to all pages.

**Reply by petergaultney** [reply]
amazing timing! thank you :)
